src/main/java/online/shop/impl/ProductDaoImpl.java

```diff
@@ -0,0 +1,45 @@
+package online.shop.impl;
+
+import java.util.List;
+import java.util.Optional;
+import online.shop.dao.ProductDao;
+import online.shop.db.Storage;
+import online.shop.model.Product;
+import online.shop.service.lib.Dao;
+
+@Dao
+public class ProductDaoImpl implements ProductDao {
+    @Override
+    public Product create(Product product) {
+        Storage.addProduct(product);
+        return product;
+    }
+
+    @Override
+    public Optional<Product> getById(Long productId) {
+        return Storage.products.stream()
+                .filter(id -> id.getId().equals(productId))
```

Comment:
I think you filter through products & get ID of a product, so maybe 'id' should be 'product' here?